code/tests/unittests/test_vector3.cpp

```diff
@@ -163,10 +164,55 @@ TEST_P(Vector3ParameterizedTest, BracketOperatorWhenPassingAnIndexReturnsDoubleR
 }
 
 INSTANTIATE_TEST_SUITE_P(
-    Vector3BracketOperators,
-    Vector3ParameterizedTest,
-    ::testing::Values(
-        std::make_tuple(0, 4.0), // first -> index, second -> value
+    Vector3BracketOperatorValueComparison,
+    Vector3BracketOperatorTest,
+    Values(
+        // first -> index, second -> value
+        std::make_tuple(0, 4.0),
         std::make_tuple(1, 5.0),
         std::make_tuple(2, 6.0)
     ));
+
+class Vector3RandomTest : public TestWithParam<std::tuple<double, double>> {
+protected:
+    Vector3 vector_generator;
+};
+
+TEST(Vector3RandomTest, RandomWhenNoParametersComponentIsMoreThanZeroUpToOOne) {
+    auto result = Vector3::Random();
+    ASSERT_THAT(result.e, Each(AllOf(Ge(0), Lt(1))));
+}
+
+TEST(Vector3RandomTest, RandomWhenBadMinMaxThrowError) {
+    auto min = 2;
+    auto max = 1;
+    ASSERT_THROW(Vector3::Random(min, max), std::invalid_argument);
+}
+
+TEST(Vector3RandomTest, RandomWhenGoodMinMaxThrowNoError) {
+    auto min = 1;
+    auto max = 2;
+    ASSERT_NO_THROW(Vector3::Random(min, max));
+}
+
+TEST(Vector3RandomTest, RandomInUnitSphereNoParametersVectorIsInsideUnitSphere) {
+    auto result = RandomInUnitSphere();
+    ASSERT_LT(result.LengthSquared(), 1);
+}
+
+TEST_P(Vector3RandomTest, RandomWhenMinMaxComponentIsMoreThanMinUpToMax) {
```

Comment:
We could change this to RandomWhenPassingMinMaxParametersReturnsVectorWhithinRange